File: AD-HOC/1437/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] directions = { "N", "L", "S", "O" };
        while (scanner.nextInt() != 0) {
            int direction = 0;
            for (String command : scanner.next().split(""))
                direction = command.equals("D") ? direction + 1 : direction - 1;
            int last = (direction % 4 + 4) % 4;
            System.out.println(directions[last]);
        }
        scanner.close();
    }
}

File: AD-HOC/1437/main.c
#include <stdio.h>

int main() {
    char directions[4] = {'N', 'L', 'S', 'O'};
    int n;
    scanf("%d", &n);
    while (n) {
        int direction = 0;
        char commands[n];
        scanf("%s", &commands);
        for (int i = 0; i < n; i++)
            commands[i] == 'D' ? direction++ : direction--;
        int last = (direction % 4 + 4) % 4;
        printf("%c\n", directions[last]);
        scanf("%d", &n);
    }
    return 0;
}
